### DISCUSSION TITLE
Welcome to u8g2 Discussions!

### DISCUSSION THREAD

**Discussion by olikraus** [olikraus/u8g2#1356] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by modarsoos** [comment]
Hello,
I'm Modar From Germany
I love your library and want to make more development on it, adding new more led to be the only one library on the net
thank you for your programing

**Comment by CodeKek** [comment]
Hi,
 very mighty library! Thanks a lot!

I need to mount a display rotated by 90 degrees. I assume, there is no function to do so. 
My idea is to add a new font in "u8g2_fonts.c"
But I have no idea about the coding of the font data ("_\3\4\3\4\4\1\1\5\10\7\0\0\10\0\10\1\1~\3 \4\207 \7\210\343\307\21\0!\11")
Could you please explain?

Thank you very much!

Peter

**Reply by AnHardt** [reply]
Your assumption is wrong. Have a look at the constructors first parameter. Replace `U8G2_R0` by `U8G2_R1` or `U8G2_R3`. [RTFM](https://github.com/olikraus/u8g2/wiki/u8g2reference)

**Reply by CodeKek** [reply]
Thanks a lot! Great!!!

**Reply by CodeKek** [reply]
Problem still unsolved:
I need R1 & MIRROR. 
Any idea? 
As I need only numbers an a handful of chars - drawXBM?

**Reply by olikraus** [reply]
Well, yes R1&Mirror is not supported, but why do you need "MIRROR"?

**Comment by CodeKek** [comment]
I want to use a tiny display in an old SLR camera viewfinder. Due to the pentaprism the image is mirrored.

But no problem, I just defined the numbers and some letters as bitmaps.

 
 

Gesendet: Montag, 21. Dezember 2020 um 18:53 Uhr
Von: "olikraus" <notifications@github.com>
An: "olikraus/u8g2" <u8g2@noreply.github.com>
Cc: "vwpeter" <vw-peter@gmx-topmail.de>, "Comment" <comment@noreply.github.com>
Betreff: Re: [olikraus/u8g2] Welcome to u8g2 Discussions! (#1356)


 

Well, yes R1&Mirror is not supported, but why do you need "MIRROR"?

**Comment by xroo** [comment]
Hello. I made a russification of Nokia22 font for my needs. I'm happy to share the result.
[https://drive.google.com/file/d/1P3i9pW8FVodFrBD2_c86hJ5tUKuwN6c0/view?usp=sharing](url)
It will be great if there will be more cyrillic fonts in differrent sizes.

**Comment by shiroto1214** [comment]
Is there a way to create a font? .

Thank you for creating a great library for GLCD and for using it to some extent (for your own purposes only).
Now, I wanted to use this library to display Japanese Kana and English characters on the GLCD.
The font size is mainly 16x16dot and 32x32dot.
However, the U8g2 font had English characters of this size, but not Japanese Kana characters.
So I created the fonts I needed in XBM to suit my needs.
However, they can only be used as graphic characters, not as fonts.
Register the character code 0-127 as an alphabetic character and 128-255 as a kana character 
(however, do not use all the codes, but use some common codes (in Japanese char_LCD) and add them to the end.)
 I want to use it as same tr_ font for characters, but I don't know how to register it in the font.
Is there a way to create a font or use an existing graphic character as a font in the u8g2 library?
I can't handle it with too difficult law, but please let me know if there is an easy way to register as a e-mail or your home-page.
If there is no way to do that, use it as a graphic character.
Thank you for your guidance.

**Reply by olikraus** [reply]
U8g2 will use bdf font format. Once the bdf font is available, it can be converted to a "C" array of unsigned char's. This can be pasted into your code and used as argument for the u8g2.setFont command.
The u8g2 FAQ also includes several answers here:
https://github.com/olikraus/u8g2/blob/3c6460a73f7f310c665cef0af1d2bac49bf6c655/doc/faq.txt#L233-L251

Additionally, I once wrote a small tool which can generate a bdf file from several png files: https://github.com/olikraus/u8g2/tree/master/tools/font/png2bdf
This means: If you glyphs are available as xbm you could just convert them to png and feed those files into png2bdf. Then use the generated bdf file to generate the U8g2 font.

**Reply by shiroto1214** [reply]
Thank you for your reply.
There are some facts that go beyond my understanding, but I'll take the time to do it. (Because I am 73 years old.)
Thank you for showing me how.

**Comment by MaartenSneep** [comment]
I'm looking into the Raspberry Pi Pico, and specifically using it with a display. Instead of rolling my own, it seems a shame to at least not try to reuse the U8g2 library. Do you have suggestions on where to start for such a port?

**Reply by olikraus** [reply]
A user once contributed some code: https://github.com/olikraus/u8g2/tree/master/sys/linux-i2c
I have never used this code nor can I support this, but maybe it's a starting point.

**Reply by olikraus** [reply]
Some more hints might be here:
https://github.com/olikraus/u8g2/wiki/Porting-to-new-MCU-platform#raspberry-pi

**Reply by MaartenSneep** [reply]
Thanks, I'll (eventually) have a look. The Pico is a microcontroller, not a ARM-linux Raspberry Pi computer. But there may be some useful bits in there.

**Reply by olikraus** [reply]
Ah, ok. In general this page should give you more generic advice: https://github.com/olikraus/u8g2/wiki/Porting-to-new-MCU-platform
The repo also contains several ARM projects: https://github.com/olikraus/u8g2/tree/master/sys/arm

**Comment by jandemna** [comment]
Hi, I am Jan from NL
I like programming Arduinos and have attached to my D1 Mini an 1.3 inch Oled display for displaying time, weather and KAKU control.
The OLED is sh1106 SPI and works fine. Ug82 gives me nice fonts and symbols!
Problem I have is that the display slowly changes intensity from low to very bright. Cycle takes about 6s. Sometimes after new power up this not the case, but very often it is unfortunately. Looked at low level stuff already, but I am not an expert there. To me it look like "something" is not initialized and likes to come on different after power up. Any ideas welcome!
![2021-04-18 12 14 32](https://user-images.githubusercontent.com/34832939/115153260-468a0400-a075-11eb-90da-5a1318ac740e.jpg)
